### PR TITLE
Blackout qol

### LINKED ISSUE #79138: "is blacking out" appearing often with no name or associated ghost role popup
Reporting client version: 514.1589

## Round ID:
[217197](https://scrubby.melonmesa.com/round/217197)

## Testmerges:
- [Makes lighting objects objects again](https://www.github.com/tgstation/tgstation/pull/78857)

## Reproduction:

I have noticed this appearing in chat on random intervals across several rounds. There is no associated popup offering ghosts to take over as the blacked out split personality.

I did an SDQL2 "SELECT /datum/brain_trauma/severe/split_personality/blackout" after this happened and the only instance of this brain trauma was in a brain sitting on the floor of the Cyborg Mothership space ruin. No fingerprints, looks like it spawned there.

### PR DESCRIPTION
**Pull request by TheBoondock**
## About The Pull Request
This pr improves the gameplay of blackout drunk character via:
-Add hiccup sound cues and emote
-Add warning for duration running out
-Makes it more reasonable for blackout character to survives
-Makes it harder for blackout character to sleep
Balance changes:
-Blackout char will have random 10 seconds bouts of inability to interact with advance tools, computer, tools, scanner etc..
-Blackout char will gain some stamina regen, sort of a "too drunk to feel tired" buff
To-do
- [x] Add some sort of buff for blackout character
## Why It's Good For The Game
Makes it so blackout character gameplay is more enjoyable than just getting sleep for 90 seconds, plus more feedback for the surrounding player to know that they are blackout. The penalty is added so that its harder for player to majorly grief i.e assembling the ttv, or turning off power/sm cooling, wont stop them outright tho.
Fixes #79138, #79163
## Changelog
:cl:
qol: improves blackout drunk character gameplay
fix: fixed improper prob() placement that caused blackout character to be forced sleep
sound: added hiccup sound
/:cl:

### CODE REVIEW

**Comment by EricZilla** [comment]
You should make them waddle like the clown too lmao

**Comment by Wallemations** [comment]
You could increase their wound resistance, since I don't think we have any form of pain in the code

**Comment by TheBoondock** [comment]
> You could increase their wound resistance, since I don't think we have any form of pain in the code

Staminadmg is sort of our pain system so thats what im giving em

File: code/datums/brain_damage/split_personality.dm [hunk 1]
```diff
@@ -275,6 +277,18 @@
...
 		return
+	else if(duration_in_seconds <= 50)
+		to_chat(owner, span_warning("You have 50 seconds left before sobering up!"))
+	if(prob(20) && HAS_TRAIT(brain, TRAIT_ADVANCEDTOOLUSER))
+		REMOVE_TRAIT(brain, TRAIT_ADVANCEDTOOLUSER, ORGAN_TRAIT)
+		owner.balloon_alert(owner, "dexterity reduced temporarily!")
+		//We then send a callback to automatically re-add the trait
+		addtimer(TRAIT_CALLBACK_ADD(brain, TRAIT_ADVANCEDTOOLUSER, ORGAN_TRAIT), 10 SECONDS)
+		addtimer(CALLBACK(owner, TYPE_PROC_REF(/atom, balloon_alert), owner, "dexterity regained!"), 10 SECONDS)
+	if(prob(30))
```
Comment (Ghommie) [inline code/datums/brain_damage/split_personality.dm:288]: Might be hiccupping a tad too frequently.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: sound/effects/sf_hiccup_male_01.ogg
[the remark this answers is not in the document]
Comment (TheBoondock) [reply sound/effects/sf_hiccup_male_01.ogg:1]: liscense is CC BY 3

[the remark this answers is not in the document]
Comment (Ghommie) [reply sound/effects/sf_hiccup_male_01.ogg:1]: Then include it in the attribution file.

File: code/datums/status_effects/debuffs/drunk.dm [hunk 2]
```diff
@@ -192,11 +192,13 @@
 /datum/status_effect/inebriated/drunk/proc/attempt_to_blackout()
 	var/mob/living/carbon/drunkard = owner
 	if(drunkard.gain_trauma(/datum/brain_trauma/severe/split_personality/blackout, TRAUMA_LIMIT_ABSOLUTE))
-		drunk_value -= 50 //So that the drunk personality can spice things up without being killed by liver failure
+		drunk_value -= 70 //So that the drunk personality can spice things up without being killed by liver failure
+		return
+	else if(drunkard.has_trauma_type(/datum/brain_trauma/severe/split_personality/blackout))
+		if(prob(10))
+			to_chat(owner, span_warning("You stumbled and fall over!"))
+			owner.slip(1 SECONDS)
```
Comment (Ghommie) [inline code/datums/status_effects/debuffs/drunk.dm:200]: Hey, can you move this check before the gain_trauma line? That may fix #79138, but at the least we won't constantly attempt to give the trauma when the mob already has it. Also remove the `else` from the `if` control statement, it's unnecessary around `return` statements.

File: sound/effects/sf_hiccup_male_01.ogg
[the remark this answers is not in the document]
Comment (TheBoondock) [reply sound/effects/sf_hiccup_male_01.ogg:1]: oops sorry the producers uses a unique liscense which im sure is available under section 4 of their aggreement https://www.soundfishing.eu/licence-LESF.html

**Pull request merged**